device/src/region/fixed_channel_plans/mod.rs

```diff
@@ -8,15 +8,71 @@ mod us915;
 pub(crate) use au915::AU915;
 pub(crate) use us915::US915;
 
+#[derive(Clone)]
+struct PreferredJoinChannels {
+    channel_list: heapless::Vec<u8, 16>,
+    // Decrementing number representing how many tries we have left with the specified list before
+    // reverting to the default channel selection behavior.
+    num_retries: usize,
+}
+
+#[derive(Default, Clone)]
+struct JoinChannels {
+    preferred_channels: Option<PreferredJoinChannels>,
+    // List of subbands that haven't already been tried.
+    available_subbands: heapless::Vec<u8, 9>,
```

Comment:
Could this be implemented with bool flags? Start with all bits high and but them low as you try each subband.